### ISSUE TITLE
In-graph Tracing

### ISSUE THREAD

**Issue by jotak** [kiali/kiali#2677]
*Placeholder for new in-graph tracing feature / epic*

**Comment by jotak** [comment]
**Sub-tasks**

- pre-requisite / refactoring https://github.com/kiali/kiali/issues/2732

**Comment by jmazzitelli** [comment]
fyi: This is sort of related to the other effort re: operations (classified metrics) and how to graph those. See: https://github.com/kiali/kiali/issues/2856

**Comment by jotak** [comment]
Current state:
Traces list + highlight on graph + selection panel:

![Capture d’écran de 2020-07-28 16-53-21](https://user-images.githubusercontent.com/2153442/88689453-c7295e80-d0fa-11ea-8a80-30c29252c739.png)

**Comment by jotak** [comment]
@kiali/ux-reviewers @lwrigh @beaumorley , I'll try to summarize some key points (feel free to add more) :

- Design of the trace overlay (the blue-purple highlighting on graph): boldness, color, transparency ... (just thinking about it now, perhaps that we could re-use the same color that we use for traces (kind of light blue) to keep this mental association ; but unfortunately it's kind of conflicting with the blue of the graph nodes)

- Side-panel vertical split: we've already considered several places that could host this Trace information: tooltip over graph (but it makes it too dependent on mouse move), popup (but it's more intrusive and would hide part of the graph below), or a similar panel at a different position... One interesting thing that we want to keep is the ability to view at the same time this trace information and traces or traffic data (the tabs in the summary panel).

- Refresh component: by default the traces list are refreshed at the same time as the graph refresh, they are tied together. But sometimes it can be a problem, because the traces list can be renewed (upon refresh) while potentially the user was looking at it. So, we offer the possibility to "untie" the traces list from graph refresh, and hence, offer a separate refresh button. The button is disabled when "Use graph refresh" is checked (perhaps it could be hidden instead?). When enabled, it uses a different design than our usual refresh button: patternfly's "secondary" button variant (to better distinguish from main refresh button), and smaller (better aligned with side-panel small fonts & design).

- The "Go to Tracing" button, which is at the bottom of the traces list. This button brings us to the Service Traces page, out of the graph, where the user can refine search with more options. We were not totally sure about which text to put. "Show Traces" sounded weird because we are already showing traces here... "Find more traces" was also weird because it makes us think it will add more items to the list. So we've ended up with "Go to Tracing".

**Comment by jshaughn** [comment]
> Design of the trace overlay (the blue-purple highlighting on graph): boldness, color, transparency 

I agree that blue does not work well given current coloring in the graph. We could have a session just to try out some different colors/widths/opacities.  I did thin and lighten the overlay and I think it looks better:
![image](https://user-images.githubusercontent.com/2104052/88719678-ab6d8a80-d0f1-11ea-8ff1-087419aadde6.png)

> Side-panel vertical split: 

After seeing things in the meeting I think perhaps after choosing a trace maybe the entire tab panel should be devoted to the trace, and to get the list back perhaps close the panel.  If not that then think the list should take very little space, maybe 25-30% with the rest going to the trace detail.

> Refresh component: 

I wonder now whether we need only manual refresh.  Is there really a need to tie it to the graph refresh?  If we do keep the controls I think visually the option should float left and the refresh button should float right.

**Comment by jotak** [comment]
> > Side-panel vertical split:
> 
> After seeing things in the meeting I think perhaps after choosing a trace maybe the entire tab panel should be devoted to the trace, and to get the list back perhaps close the panel. If not that then think the list should take very little space, maybe 25-30% with the rest going to the trace detail.

I don't think it would be good in terms of flow, when you want to quickly see different traces:
- click one trace
- close
- click another trace
- close
- etc.

Instead of just:
- click one trace
- click another trace
- etc.

And also, we would then loose the ability to compare trace details with the content of traffic tab (e.g. to compare some spans duration with the metrics averages / p99)

**Comment by jshaughn** [comment]
> I don't think it would be good in terms of flow, when you want to quickly see different traces:

Makes sense, although I do think we could use a larger percentage for the details.

**Comment by jotak** [comment]
> I agree that blue does not work well given current coloring in the graph. We could have a session just to try out some different colors/widths/opacities. I did thin and lighten the overlay and I think it looks better:

+1 for me, PR welcome! :-)

> > Refresh component:
> 
> I wonder now whether we need only manual refresh. Is there really a need to tie it to the graph refresh? If we do keep the controls I think visually the option should float left and the refresh button should float right.

IDK, graph refresh adds the auto-refresh feature. I'm not sure how useful it is for fetching traces. I defer to the people here :-)
In the meantime I can right-align the button as you suggest.

**Comment by jotak** [comment]
All tasks done
